extension/chrome/elements/compose-modules/compose-storage-module.ts

```diff
@@ -75,28 +75,52 @@ export class ComposeStorageModule extends ViewModule<ComposeView> {
     //    work with arrays instead, like with `ContactStore.getOneWithAllPubkeys`.
     //    However, this whole fingerprint-base section seems unnecessary, we could likely
     //    remove it and the keys will be updated below using
     //    `lookupPubkeyFromKeyserversAndUpsertDb` anyway.
     //    discussion: https://github.com/FlowCrypt/flowcrypt-browser/pull/3898#discussion_r686229818
-    const [storedContact] = await ContactStore.get(undefined, [email]);
-    if (storedContact && storedContact.hasPgp && storedContact.pubkey && !storedContact.revoked) {
-      // checks if pubkey was updated, asynchronously. By the time user finishes composing,
-      //    newer version would have been updated in db.
-      // This implementation is imperfect in that, if sender didn't pull a particular pubkey
-      //    for a long time and the local pubkey has since expired, and there actually is a
-      //    newer version available on external key server, this may unnecessarily show "bad pubkey",
-      //    until next time user tries to enter recipient in the field again, which will at that point
-      //    get the updated key from db. This could be fixed by:
-      //      - either life fixing the UI after this call finishes, or
-      //      - making this call below synchronous and using the result directly
-      this.checkKeyserverForNewerVersionOfKnownPubkeyIfNeeded(storedContact).catch(Catch.reportErr);
-      return storedContact;
+    console.log(`>>>> Looking up in DB: ${email}`);
+    const storedContact = await ContactStore.getOneWithAllPubkeys(undefined, email);
+    console.log(">>>> " + (storedContact ? JSON.stringify(storedContact) : 'NOT_FOUND'));
+    if (storedContact && storedContact.sortedPubkeys.length) {
+      const results: PubKeyInfo[] = [];
+      for (const pubkey of storedContact.sortedPubkeys) {
+        if (!pubkey.revoked) {
+          // checks if pubkey was updated, asynchronously. By the time user finishes composing,
+          //    newer version would have been updated in db.
+          // This implementation is imperfect in that, if sender didn't pull a particular pubkey
+          //    for a long time and the local pubkey has since expired, and there actually is a
+          //    newer version available on external key server, this may unnecessarily show "bad pubkey",
+          //    until next time user tries to enter recipient in the field again, which will at that point
+          //    get the updated key from db. This could be fixed by:
+          //      - either life fixing the UI after this call finishes, or
+          //      - making this call below synchronous and using the result directly
+          this.checkKeyserverForNewerVersionOfKnownPubkeyIfNeeded(storedContact.info.email, pubkey)
+            .catch(Catch.reportErr);
+          results.push(pubkey);
+        } else {
+          const res = await this.lookupPubkeyFromKeyserversAndUpsertDb(email, name, pubkey);
```

Comment:
why call `lookup...()` several times for the same `email`? actually `pubkey` parameter is no longer needed as this method no longer needs to form a "contact". And this method should return PubkeyInfo[]. (please check for duplicates with existing set).

Reply:
UPD: as `lookupPubkeyFromKeyserversAndUpsertDb` re-reads the storage, this is all wrong.
We should call `lookupPubkeyFromKeyserversAndUpsertDb(email, name)` and re-read from storage